ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/BackupReplayFile.java

```diff
@@ -16,36 +16,21 @@
 package io.confluent.ksql.rest.server;
 
 import io.confluent.ksql.util.KsqlException;
-import io.confluent.ksql.util.Pair;
 import java.io.Closeable;
 import java.io.File;
 import java.io.FileNotFoundException;
 import java.io.FileOutputStream;
 import java.io.IOException;
-import java.nio.charset.StandardCharsets;
-import java.nio.file.Files;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Objects;
 import org.apache.kafka.clients.consumer.ConsumerRecord;
 
 /**
  * A file that is used by the backup service to replay command_topic commands.
  */
-public final class BackupReplayFile implements Closeable {
-  private static final String KEY_VALUE_SEPARATOR_STR = ":";
-  private static final String NEW_LINE_SEPARATOR_STR = "\n";
-
-  private static final byte[] KEY_VALUE_SEPARATOR_BYTES =
-      KEY_VALUE_SEPARATOR_STR.getBytes(StandardCharsets.UTF_8);
-  private static final byte[] NEW_LINE_SEPARATOR_BYTES =
-      NEW_LINE_SEPARATOR_STR.getBytes(StandardCharsets.UTF_8);
-
-  private final File file;
+public final class BackupReplayFile extends BackupInputFile implements Closeable {
   private final FileOutputStream writer;
 
   public BackupReplayFile(final File file) {
```

Comment:
Do we need to actually split this into two classes? `BackupReplayFile` and `BackupInputFile`

It doesn't really look like there's much difference between the two so it's not clear to me why having both is necessary.

Reply:
There's a slightly difference. The `BackupReplayFile` opens the file for writing. It fails if there are no permissions for writing. I created the `BackupInputFile` just for that, so it can open it if only read permissions are set.

Reply:
I reverted the BackupInputFile. I added two static methods to the BackupReplayFile:
```
BackupReplayFile.readOnly(file);
BackupReplayFile.writetable(file);
```